Library/Homebrew/extend/os/linux/requirements/xcode_requirement.rb

```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+require "requirement"
+
+class XcodeRequirement < Requirement
+  def message
+    <<~EOS
+      A full installation of Xcode.app is required to compile this software.
```

Comment:
I don't think this message makes sense on Linux. I'd suggest that Linux considers `depends_on :xcode` to be `depends_on :macos` instead (see `dependency_collector.rb` where this symbol translation is done).

Reply:
I was trying to fix the error message when trying to install the `xctool` formula (see the failure above), which has:
```
depends_on :macos
depends_on xcode: "7.0"
```

I could skip this PR and write:
```
depends_on :macOS
on_macos do
  depends_on xcode: "7.0"
end
```

Depending on Xcode does not necessarily mean you can build only on Mac. Some formulae can be built with cmake/make/gcc on Linux even if they are built with Xcode on Mac. So I would like to avoid writing code that says: "depends_on Xcode == depends_on macOS".

Maybe `depends_on :macOS` should be the first requirement checked in the list and have an early bail out on Linux.

Reply:
I see two options:
- `depends_on :xcode` implies `depends_on :macos`
- `depends_on :xcode` is a no-op on Linux

It sounds like the latter is better.

Reply:
> I could skip this PR and write:
> ```
> depends_on :macOS
> on_macos do
>   depends_on xcode: "7.0"
> end
> ```


If the formula already depends on macOS, why does `depends_on :xcode` need to be inside a `on_macos` block? Shouldn't `depends_on :macos` be removed as well in this case if it does indeed build on Linux?

Reply:
Good point @reitermarkus 👍🏻